### ISSUE TITLE
[D3D9]Neverwinter Nights 2: Complete(GOG) crashes when entering the first out-door stage in chapter 1

### ISSUE THREAD

**Issue by v21cesc** [doitsujin/dxvk#1487]
### Software information
Neverwinter Nights 2: Complete(GOG) crashes when entering the first out-door stage in chapter 1.
Also , some rendering issues can be found with the fire place and magic effect cast by dwarves(enemy) before the crashing.

### System information
- GPU:NVIDIA 1660ti
- Driver:440.59
- Wine version: 5.2 Lutris build
- DXVK version: 1.5 - 1.5.5

**Comment by v21cesc** [comment]
Lutris log:

Running /home/v21/.local/share/lutris/runners/wine/lutris-5.2-x86_64/bin/wine /home/v21/Games/NWN2_GOG/drive_c/GOG Games/NWN2 Complete/NWN2Launcher.exe
gamemodeauto: 
esync: up and running.
Waiting on children
Waiting on children
info:  Game: nwn2main.exe
info:  DXVK: v1.5.5
info:  Built-in extension providers:
info:    Win32 WSI
info:    OpenVR
warn:  OpenVR: Failed to locate module
info:  Enabled instance extensions:
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
warn:  D3D9: VK_FORMAT_D16_UNORM_S8_UINT -> VK_FORMAT_D24_UNORM_S8_UINT
info:  GeForce GTX 1660 Ti:
info:    Driver: 440.59.0
info:    Vulkan: 1.1.119
info:    Memory Heap[0]: 
info:      Size: 6144 MiB
info:      Flags: 0x1
info:      Memory Type[7]: Property Flags = 0x1
info:      Memory Type[8]: Property Flags = 0x1
info:    Memory Heap[1]: 
info:      Size: 11960 MiB
info:      Flags: 0x0
info:      Memory Type[0]: Property Flags = 0x0
info:      Memory Type[1]: Property Flags = 0x0
info:      Memory Type[2]: Property Flags = 0x0
info:      Memory Type[3]: Property Flags = 0x0
info:      Memory Type[4]: Property Flags = 0x0
info:      Memory Type[5]: Property Flags = 0x0
info:      Memory Type[6]: Property Flags = 0x0
info:      Memory Type[9]: Property Flags = 0x6
info:      Memory Type[10]: Property Flags = 0xe
info:  Process set as DPI aware
info:  Device properties:
info:    Device name:     : GeForce GTX 1660 Ti
info:    Driver version   : 440.59.0
info:  Enabled device extensions:
info:    VK_EXT_depth_clip_enable
info:    VK_EXT_host_query_reset
info:    VK_EXT_shader_demote_to_helper_invocation
info:    VK_EXT_shader_viewport_index_layer
info:    VK_EXT_transform_feedback
info:    VK_EXT_vertex_attribute_divisor
info:    VK_KHR_create_renderpass2
info:    VK_KHR_depth_stencil_resolve
info:    VK_KHR_draw_indirect_count
info:    VK_KHR_driver_properties
info:    VK_KHR_image_format_list
info:    VK_KHR_sampler_mirror_clamp_to_edge
info:    VK_KHR_swapchain
info:  Device features:
info:    robustBufferAccess                     : 1
info:    fullDrawIndexUint32                    : 1
info:    imageCubeArray                         : 1
info:    independentBlend                       : 1
info:    geometryShader                         : 1
info:    tessellationShader                     : 0
info:    sampleRateShading                      : 1
info:    dualSrcBlend                           : 0
info:    logicOp                                : 0
info:    multiDrawIndirect                      : 0
info:    drawIndirectFirstInstance              : 0
info:    depthClamp                             : 1
info:    depthBiasClamp                         : 1
info:    fillModeNonSolid                       : 1
info:    depthBounds                            : 1
info:    multiViewport                          : 1
info:    samplerAnisotropy                      : 1
info:    textureCompressionBC                   : 1
info:    occlusionQueryPrecise                  : 1
info:    pipelineStatisticsQuery                : 1
info:    vertexPipelineStoresAndAtomics         : 1
info:    fragmentStoresAndAtomics               : 0
info:    shaderImageGatherExtended              : 0
info:    shaderStorageImageExtendedFormats      : 1
info:    shaderStorageImageReadWithoutFormat    : 0
info:    shaderStorageImageWriteWithoutFormat   : 1
info:    shaderClipDistance                     : 1
info:    shaderCullDistance                     : 1
info:    shaderFloat64                          : 0
info:    shaderInt64                            : 0
info:    variableMultisampleRate                : 0
info:  VK_EXT_conditional_rendering
info:    conditionalRendering                   : 0
info:  VK_EXT_depth_clip_enable
info:    depthClipEnable                        : 1
info:  VK_EXT_host_query_reset
info:    hostQueryReset                         : 1
info:  VK_EXT_memory_priority
info:    memoryPriority                         : 0
info:  VK_EXT_shader_demote_to_helper_invocation
info:    shaderDemoteToHelperInvocation         : 1
info:  VK_EXT_transform_feedback
info:    transformFeedback                      : 0
info:    geometryStreams                        : 0
info:  VK_EXT_vertex_attribute_divisor
info:    vertexAttributeInstanceRateDivisor     : 1
info:    vertexAttributeInstanceRateZeroDivisor : 1
info:  Queue families:
info:    Graphics : 0
info:    Transfer : 1
info:  DXVK: Read 120 valid state cache entries
info:  DXVK: Using 5 compiler threads
info:  D3D9DeviceEx::ResetSwapChain:
info:    Requested Presentation Parameters
info:      - Width:              1920
info:      - Height:             1080
info:      - Format:             D3D9Format::A8R8G8B8
info:      - Auto Depth Stencil: true
info:                  ^ Format: D3D9Format::D24S8
info:      - Windowed:           false
info:  Presenter: Actual swap chain properties:
info:    Format:       VK_FORMAT_B8G8R8A8_UNORM
info:    Present mode: VK_PRESENT_MODE_IMMEDIATE_KHR
info:    Buffer size:  1928x1107
info:    Image count:  2
info:    Exclusive FS: 0
info:  D3D9: Setting display mode: 1920x1080@60
warn:  ConvertFormat: Unknown format encountered: D3D9Format::SSAA
info:  Game: nwn2main.exe
info:  DXVK: v1.5.5
info:  Built-in extension providers:
info:    Win32 WSI
info:    OpenVR
warn:  OpenVR: Failed to locate module
info:  Enabled instance extensions:
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
warn:  D3D9: VK_FORMAT_D16_UNORM_S8_UINT -> VK_FORMAT_D24_UNORM_S8_UINT
info:  GeForce GTX 1660 Ti:
info:    Driver: 440.59.0
info:    Vulkan: 1.1.119
info:    Memory Heap[0]: 
info:      Size: 6144 MiB
info:      Flags: 0x1
info:      Memory Type[7]: Property Flags = 0x1
info:      Memory Type[8]: Property Flags = 0x1
info:    Memory Heap[1]: 
info:      Size: 11960 MiB
info:      Flags: 0x0
info:      Memory Type[0]: Property Flags = 0x0
info:      Memory Type[1]: Property Flags = 0x0
info:      Memory Type[2]: Property Flags = 0x0
info:      Memory Type[3]: Property Flags = 0x0
info:      Memory Type[4]: Property Flags = 0x0
info:      Memory Type[5]: Property Flags = 0x0
info:      Memory Type[6]: Property Flags = 0x0
info:      Memory Type[9]: Property Flags = 0x6
info:      Memory Type[10]: Property Flags = 0xe
info:  Process set as DPI aware
info:  Presenter: Actual swap chain properties:
info:    Format:       VK_FORMAT_B8G8R8A8_UNORM
info:    Present mode: VK_PRESENT_MODE_FIFO_KHR
info:    Buffer size:  1920x1080
info:    Image count:  3
info:    Exclusive FS: 0
2020-03-01 01:39:07,955: Unable to read stat for process 56546
2020-03-01 01:39:07,955: Unable to read stat for process 56546
Waiting on children
warn:  D3D9DeviceEx::SetRenderState: Unhandled render state D3DRS_MULTISAMPLEANTIALIAS
wine: Unhandled page fault on read access to 00070002 at address 00070002 (thread 003b), starting debugger...
Unhandled exception: page fault on read access to 0x00070002 in 32-bit code (0x00070002).
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:006b
 EIP:00070002 ESP:0032f378 EBP:00000000 EFLAGS:00010202(  R- --  I   - - - )
 EAX:00070002 EBX:ffffffff ECX:0500e3f8 EDX:0032f3bc
 ESI:05030338 EDI:0032f55c
Stack dump:
0x0032f378:  0044f488 05030338 0032f3bc 138193a0
0x0032f388:  00000000 00000006 00000006 0000a604
0x0032f398:  0032f3bc ffffffff 138193a0 0032f55c
0x0032f3a8:  00000000 0045d946 ffffffff 0045d959
0x0032f3b8:  00000000 05030338 0032f6b0 128d25d0
0x0032f3c8:  0032f62c 085cd510 00000000 009425f0
Backtrace:
=>0 0x00070002 (0x00000000)
0x00070002: addb	%al,0x0(%eax)
Modules:
Module	Address			Debug info	Name (68 modules)
PE	  400000-  b31000	Deferred        nwn2main
PE	10000000-10015000	Deferred        nwn2_memorymgr
PE	21100000-21164000	Deferred        mss32
PE	22300000-22338000	Deferred        msseax.m3d
PE	22400000-22419000	Deferred        msssoft.m3d
PE	22600000-22618000	Deferred        mssdx7.m3d
PE	22700000-22776000	Deferred        mssrsx.m3d
PE	24100000-24120000	Deferred        mssdsp.flt
PE	26400000-26439000	Deferred        mssvoice.asi
PE	26f00000-26f2e000	Deferred        mssmp3.asi
PE	30000000-3006d000	Deferred        binkw32
PE	50000000-50090000	Deferred        granny2
PE	62440000-626fa000	Deferred        d3d9
PE	6f400000-6f5ff000	Deferred        dxgi
PE	7a840000-7a849000	Deferred        opengl32
PE	7b020000-7b038000	Deferred        kernelbase
PE	7b420000-7b5f1000	Deferred        kernel32
PE	7bc30000-7bc47000	Deferred        ntdll
PE	7c6d0000-7c6d6000	Deferred        uxtheme
PE	7c950000-7d24c000	Deferred        shell32
PE	7d560000-7d579000	Deferred        setupapi
PE	7d790000-7d795000	Deferred        winex11
PE	7d860000-7d86e000	Deferred        oleaut32
PE	7d970000-7d97a000	Deferred        aclui
PE	7d990000-7d996000	Deferred        shcore
PE	7d9c0000-7d9c8000	Deferred        winevulkan
PE	7da00000-7da08000	Deferred        vulkan-1
PE	7da40000-7da68000	Deferred        ucrtbase
PE	7db30000-7db35000	Deferred        d3dxof
PE	7db70000-7db75000	Deferred        d3dcompiler_47
PE	7dbf0000-7dbfa000	Deferred        d3dx9_30
PE	7dc90000-7dc96000	Deferred        imm32
PE	7dcb0000-7dcb4000	Deferred        usp10
PE	7dd10000-7dd66000	Deferred        comctl32
PE	7de60000-7de6c000	Deferred        dinput8
PE	7dec0000-7ded2000	Deferred        shlwapi
PE	7df40000-7df59000	Deferred        msvcr80
PE	7e170000-7e177000	Deferred        ws2_32
PE	7e1c0000-7e1cd000	Deferred        rpcrt4
PE	7e270000-7e29d000	Deferred        ole32
PE	7e3d0000-7e4e8000	Deferred        dbghelp
PE	7e3d0000-7e4e8000	Deferred        msvcrt
PE	7e520000-7e525000	Deferred        version
PE	7e540000-7e54d000	Deferred        advapi32
PE	7e5d0000-7e5de000	Deferred        gdi32
PE	7e730000-7e824000	Deferred        user32
PE	7e950000-7e9d0000	Deferred        winmm
PE	7ea10000-7ea13000	Deferred        wow64cpu
PE	d7740000-d7749000	Deferred        windowscodecs
PE	d7e40000-d7e4b000	Deferred        quartz
PE	d8020000-d8028000	Deferred        propsys
PE	d8280000-d8284000	Deferred        midimap
PE	d88c0000-d88c4000	Deferred        msacm32
PE	d88e0000-d88e4000	Deferred        winealsa
PE	d8920000-d892d000	Deferred        msvfw32
PE	d8950000-d8955000	Deferred        devenum
PE	dbdb0000-dbdb5000	Deferred        msdmo
PE	dc530000-dc537000	Deferred        iphlpapi
PE	dc560000-dc56e000	Deferred        winspool
PE	dc5b0000-dc5b5000	Deferred        wbemprox
PE	dc630000-dc63b000	Deferred        wined3d
PE	dc7a0000-dc7a5000	Deferred        ddraw
PE	dc920000-dc926000	Deferred        dxdiagn
PE	f1e10000-f1e14000	Deferred        avicap32
PE	f6080000-f6084000	Deferred        winepulse
PE	f60c0000-f60c5000	Deferred        mmdevapi
PE	f60f0000-f60f5000	Deferred        dsound
PE	f6140000-f614a000	Deferred        msacm32
Threads:
process  tid      prio (all id:s are in hex)
0000000e services.exe
	00000025    0
	00000022    0
	0000001d    0
	0000001a    0
	00000013    0
	00000010    0
	0000000f    0
00000011 plugplay.exe
	00000017    0
	00000016    0
	00000012    0
00000018 winedevice.exe
	0000001f    0
	0000001e    0
	0000001c    0
	0000001b    0
	00000019    0
00000020 winedevice.exe
	00000026    0
	00000024    0
	00000023    0
	00000021    0
00000027 explorer.exe
	0000002b    0
	0000002a    0
	00000029    0
	00000028    0
00000038 nwn2.exe
	[nwn2.exe]
	00000039    0
0000003a (D) C:\GOG Games\NWN2 Complete\nwn2main.exe
	[nwn2main.exe ]
	00000060    0
	0000005d    0
	0000005c    0
	0000005b    0
	00000056    0
	00000055    0
	00000054    0
	00000052    0
	00000051    0
	00000050    0
	0000004f    0
	0000004e   -2
	0000004d   -2
	0000004c   -2
	0000004b   -2
	0000004a   -2
	00000047   15
	00000046    0
	00000043   15
	0000003e   15
	0000003d    0
	0000003c    0
	0000003b    0 <==
System information:
    Wine build: wine-5.2.r1.g8b2fd051 ( TkG Staging Esync Fsync )
    Platform: i386 (WOW64)
    Version: Windows 7
    Host system: Linux
    Host version: 5.5.5-200.fc31.x86_64
2020-03-01 01:42:30,133: Unable to read stat for process 56548
2020-03-01 01:42:30,133: Unable to read stat for process 56548
Waiting on children
2020-03-01 01:42:30,436: Unable to read stat for process 56552
2020-03-01 01:42:30,437: Unable to read stat for process 56552
All children gone
Exit with returncode 0

**Comment by WinterSnowfall** [comment]
> Neverwinter Nights 2: Complete(GOG) crashes when entering the first out-door stage in chapter 1.

Do you mean the West Harbor exterior?

![nwn2](https://user-images.githubusercontent.com/6306593/75628568-fc309400-5be2-11ea-9f78-2bec79385ae4.jpg)

**Comment by v21cesc** [comment]
> > Neverwinter Nights 2: Complete(GOG) crashes when entering the first out-door stage in chapter 1.
> 
> Do you mean the West Harbor exterior?
> 
> ![nwn2](https://user-images.githubusercontent.com/6306593/75628568-fc309400-5be2-11ea-9f78-2bec79385ae4.jpg)

Yeah, It's West harbor indeed. How did you config this game?

**Comment by WinterSnowfall** [comment]
> Yeah, It's West harbor indeed. How did you config this game?

Nothing special, albeit I'm using pure Wine Staging without Lutris, which does not have any of the esync/fsync stuff. Perhaps try disabling those?

In any case, it's most likely not a DXVK problem. I'm also not getting any issue with the fireplace or magic effects at the start.

**Comment by v21cesc** [comment]
> > Yeah, It's West harbor indeed. How did you config this game?
> 
> Nothing special, albeit I'm using pure Wine Staging without Lutris, which does not have any of the esync/fsync stuff enabled by default. Perhaps try disabling those?
> 
> In any case, it's most likely not a DXVK problem. I'm also not getting any issue with the fireplace or magic effects at the start.

Well,I finally fix the issue mentioned above by additionally installing dx9 runtime lib to nwn2's wine prefix path though it is strongly recommend not to do this. And I'm afraid that this issue could be caused by a d3d9/d3d9x function missing of DXVK.  
What kind of log file can I supply for further details?

**Comment by Joshua-Ashton** [comment]
We don't implement d3dx9. Please make an issue on the Wine bug tracker.

**Comment by WinterSnowfall** [comment]
> Well,I finally fix the issue mentioned above by additionally installing dx9 runtime lib to nwn2's wine prefix path though it is strongly recommend not to do this.

Ah, makes sense. Sorry, I forgot to mention I've used winetricks to get the native versions of all d3dcompiler and d3dx9 dlls installed and overridden, but that's generally recommended with DXVK.

> PE 7db70000-7db75000 Deferred d3dcompiler_47
> PE 7dbf0000-7dbfa000 Deferred d3dx9_30

I'm guessing you need the native versions of the above for it to work properly, not necessarily everything.

If you're unhappy about using the native versions for whatever reason, as Josh said, you can raise an issue with the Wine devs.

**Comment by v21cesc** [comment]
> > Well,I finally fix the issue mentioned above by additionally installing dx9 runtime lib to nwn2's wine prefix path though it is strongly recommend not to do this.
> 
> Ah, makes sense. Sorry, I forgot to mention I've used winetricks to get the native versions of all d3dcompiler and d3dx9 dlls installed and overridden, but that's generally recommended with DXVK.
> 
> > PE 7db70000-7db75000 Deferred d3dcompiler_47
> > PE 7dbf0000-7dbfa000 Deferred d3dx9_30
> 
> I'm guessing you need the native versions of the above for it to work properly, not necessarily everything.
> 
> If you're unhappy about using the native versions for whatever reason, as Josh said, you can raise an issue with the Wine devs.

I don't mind at all. The game runs perfectly now.

**Comment by v21cesc** [comment]
> > Well,I finally fix the issue mentioned above by additionally installing dx9 runtime lib to nwn2's wine prefix path though it is strongly recommend not to do this.
> 
> Ah, makes sense. Sorry, I forgot to mention I've used winetricks to get the native versions of all d3dcompiler and d3dx9 dlls installed and overridden, but that's generally recommended with DXVK.
> 
> > PE 7db70000-7db75000 Deferred d3dcompiler_47
> > PE 7dbf0000-7dbfa000 Deferred d3dx9_30
> 
> I'm guessing you need the native versions of the above for it to work properly, not necessarily everything.
> 
> If you're unhappy about using the native versions for whatever reason, as Josh said, you can raise an issue with the Wine devs.

Alright Joshua, I'll close this thread
